agent/stats/reporter/reporter.go

```diff
@@ -48,22 +60,20 @@ func NewDockerTelemetrySession(
 	taskEngine engine.TaskEngine,
 	metricsChannel <-chan ecstcs.TelemetryMessage,
 	healthChannel <-chan ecstcs.HealthMessage,
-	doctor *doctor.Doctor) *DockerTelemetrySession {
+	doctor *doctor.Doctor) (*DockerTelemetrySession, error) {
 	ok, cfgParseErr := isContainerHealthMetricsDisabled(cfg)
 	if cfgParseErr != nil {
-		seelog.Warnf("Error starting metrics session: %v", cfgParseErr)
-		return nil
+		logger.Warn("Error starting metrics session", logger.Fields{
+			field.Error: cfgParseErr,
+		})
+		return nil, cfgParseErr
 	}
 	if ok {
-		seelog.Warnf("Metrics were disabled, not starting the telemetry session")
-		return nil
+		logger.Warn("Metrics were disabled, not starting the telemetry session")
+		return nil, nil
 	}
 
 	agentVersion, agentHash, containerRuntimeVersion := generateVersionInfo(taskEngine)
-	if cfg == nil {
```

Comment:
Just want to confirm, line 63-66 are removed because function `isContainerHealthMetricsDisabled()` will return an error if cfg is empty/nil. And this error will be caught and returned in line 65-69.

Reply:
Yes that's correct.